tests/test_minimizer.py

```diff
@@ -209,9 +217,56 @@ def test_local_minimize_water_box():
     np.testing.assert_array_equal(g_init, g_init_test)
 
 
+def test_local_minimize_water_box_with_bounds():
+    """
+    Test that we can locally relax a box of water using L-BFGS-B with bounds
+    """
+    ff = Forcefield.load_default()
+
+    system, x0, box0, top = builders.build_water_system(4.0, ff.water_ff)
+    host_fns, _ = openmm_deserializer.deserialize_system(system, cutoff=1.2)
+    box0 += np.diag([0.1, 0.1, 0.1])  # remove any possible clashes at the boundary
+
+    val_and_grad_fn = minimizer.get_val_and_grad_fn(host_fns, box0)
+
+    free_idxs = [0, 2, 3, 6, 7, 9, 15, 16]
+    frozen_idxs = set(range(len(x0))).difference(set(free_idxs))
+    frozen_idxs = list(frozen_idxs)
+
+    allowed_diff = 0.01
+
+    # Set up the bounds
+    lower_bounds = np.array(x0[free_idxs].reshape(-1)) - allowed_diff
+    upper_bounds = np.array(x0[free_idxs].reshape(-1)) + allowed_diff
+    bounds = list(zip(lower_bounds, upper_bounds))
+
+    minimizer_config = minimizer.ScipyMinimizationConfig("L-BFGS-B", bounds=bounds)
+
+    u_init, g_init = val_and_grad_fn(x0)
+
+    x_opt = minimizer.local_minimize(x0, box0, val_and_grad_fn, free_idxs, minimizer_config)
+
+    np.testing.assert_array_equal(x0[frozen_idxs], x_opt[frozen_idxs])
+    assert 0.01 < np.linalg.norm(x0[free_idxs] - x_opt[free_idxs]) <= np.linalg.norm([allowed_diff * 2] * 3)
```

Comment:
nit: worth strengthening to assert that `x_opt` is within the bounds?

Reply:
All of the `x_opt` values are identical besides `free_idxs`, as tested above. Wouldn't strengthen it to do `x0 - x_opt` if I am understanding what you are suggesting

Reply:
Ah makes sense, will do.

Reply:
Strengthened in https://github.com/proteneer/timemachine/pull/1405/commits/5e128eacf3cbf43324c59b488a8733a4030a1058